daemon/algod/api/server/v2/delta.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package v2
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/daemon/algod/api/server/v2/generated/model"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+)
+
+// convertAppResourceRecordToGenerated takes ledgercore.AppResourceRecord and converts it to v2.model.AppResourceRecord
+func convertAppResourceRecordToGenerated(app ledgercore.AppResourceRecord) model.AppResourceRecord {
+	var appLocalState *model.ApplicationLocalState = nil
+	if app.State.LocalState != nil {
+		localState := convertTKVToGenerated(&app.State.LocalState.KeyValue)
```

Comment:
@Eric-Warehime There are several places where the PR duplicates conversion logic.  Unless there's a by-design reason to duplicate the logic, can you review https://github.com/Eric-Warehime/go-algorand/pull/5?

The child PR attempts to remove conversion duplication.